pkg/meta/dynamodb/dynamo.go

```diff
@@ -1363,195 +1145,138 @@ func (dwr *DynamoDB) FilterRepos(ctx context.Context,
 		repoMeta.IsStarred = zcommon.Contains(userStars, repoMeta.Name)
 
 		if filter(repoMeta) {
-			pageFinder.Add(mTypes.DetailedRepoMeta{
-				RepoMetadata: repoMeta,
-			})
+			foundRepos = append(foundRepos, repoMeta)
 		}
 	}
 
-	foundRepos, pageInfo := pageFinder.Page()
-
 	foundManifestMetadataMap, foundIndexDataMap, err := common.FetchDataForRepos(dwr, foundRepos)
 
-	return foundRepos, foundManifestMetadataMap, foundIndexDataMap, pageInfo, err
+	return foundRepos, foundManifestMetadataMap, foundIndexDataMap, err
 }
 
-func (dwr *DynamoDB) SearchTags(ctx context.Context, searchText string, filter mTypes.Filter,
-	requestedPage mTypes.PageInput,
+func (dwr *DynamoDB) SearchTags(ctx context.Context, searchText string,
 ) ([]mTypes.RepoMetadata, map[string]mTypes.ManifestMetadata, map[string]mTypes.IndexData,
-	zcommon.PageInfo, error,
+	error,
 ) {
 	var (
+		foundRepos                = make([]mTypes.RepoMetadata, 0, 1)
 		manifestMetadataMap       = make(map[string]mTypes.ManifestMetadata)
 		indexDataMap              = make(map[string]mTypes.IndexData)
 		repoMetaAttributeIterator AttributesIterator
-		pageFinder                pagination.PageFinder
-		pageInfo                  zcommon.PageInfo
 		userBookmarks             = getUserBookmarks(ctx, dwr)
 		userStars                 = getUserStars(ctx, dwr)
 	)
 
-	pageFinder, err := pagination.NewBaseImagePageFinder(requestedPage.Limit, requestedPage.Offset, requestedPage.SortBy)
-	if err != nil {
-		return []mTypes.RepoMetadata{}, map[string]mTypes.ManifestMetadata{}, map[string]mTypes.IndexData{},
-			pageInfo, err
-	}
-
 	repoMetaAttributeIterator = NewBaseDynamoAttributesIterator(
 		dwr.Client, dwr.RepoMetaTablename, "RepoMetadata", 0, dwr.Log,
 	)
 
 	searchedRepo, searchedTag, err := common.GetRepoTag(searchText)
 	if err != nil {
 		return []mTypes.RepoMetadata{}, map[string]mTypes.ManifestMetadata{}, map[string]mTypes.IndexData{},
-			pageInfo,
 			fmt.Errorf("metadb: error while parsing search text, invalid format %w", err)
 	}
 
 	repoMetaAttribute, err := repoMetaAttributeIterator.First(ctx)
+	if err != nil {
+		return []mTypes.RepoMetadata{}, map[string]mTypes.ManifestMetadata{}, map[string]mTypes.IndexData{},
+			err
+	}
 
-	for ; repoMetaAttribute != nil; repoMetaAttribute, err = repoMetaAttributeIterator.Next(ctx) {
```

Comment:
Why did we remove this?

Reply:
Because we assume the input of search tags contains "repo:tag" with a valid repo. So search tags should always return only 0 or maximum 1 RepoMeta objects.

Reply:
Understood.